compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2958,7 +2958,6 @@ private void assignTypesToMemberPatterns(BLangMatchPattern matchPattern, BType b
                     for (BType type : ((BUnionType) patternType).getMemberTypes()) {
                         assignTypesToMemberPatterns(listMatchPattern, type, data);
                     }
-                    listMatchPattern.setBType(patternType);
```

Comment:
This change leads o fail three test cases. Currently looking into it. Therefore mark this PR as draft